File: Java/Medium/WrongAnswer/append-k-integers-with-minimal-sum_4.java
class Solution {
    public long minimalKSum(int[] nums, int k) {
        
        long sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
        }
        Arrays.sort(nums);
        int i = 0;
        while (k > 0) {
            if (i == nums.length - 1) {
                sum += k;
                break;
            }
            if (nums[i + 1] - nums[i] > 1) {
                int count = Math.min(k, nums[i + 1] - nums[i] - 1);
                sum += count * (nums[i] + 1);
                k -= count;
            }
            i++;
        }
        return sum;
    }
}

File: C++/Medium/WrongAnswer/append-k-integers-with-minimal-sum_4.cpp
class Solution {
public:
    long long minimalKSum(vector<int>& nums, int k) {
        
        long long sum = 0;
        sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];
        }
        int last = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] - last > 1) {
                int add = min(nums[i] - last - 1, k);
                sum += add * (last + 1) + (add * (add - 1)) / 2;
                k -= add;
                if (k == 0) {
                    return sum;
                }
            }
            last = nums[i];
        }
        sum += k * (last + 1) + (k * (k - 1)) / 2;
        return sum;
    }
};